gssapi.go

```diff
@@ -0,0 +1,130 @@
+package ldap
+
+import (
+	"fmt"
+
+	"github.com/jcmturner/gokrb5/v8/client"
+	"github.com/jcmturner/gokrb5/v8/crypto"
+	"github.com/jcmturner/gokrb5/v8/iana/keyusage"
+	"github.com/jcmturner/gokrb5/v8/messages"
+	"github.com/jcmturner/gokrb5/v8/types"
+
+	"github.com/jcmturner/gokrb5/v8/gssapi"
+	"github.com/jcmturner/gokrb5/v8/spnego"
+)
+
+type GSSAPIState struct {
+	AuthZID string
+	token   spnego.KRB5Token
+	ekey    types.EncryptionKey
+	Subkey  types.EncryptionKey
+	init    bool
+	asrep   bool
+}
+
+func InitContext(client *client.Client, principal, AuthZID string) (*GSSAPIState, error) {
+	tkt, ekey, err := client.GetServiceTicket(principal)
+	if err != nil {
+		return nil, err
+	}
+
+	token, err := spnego.NewKRB5TokenAPREQ(client, tkt, ekey, []int{gssapi.ContextFlagInteg, gssapi.ContextFlagConf, gssapi.ContextFlagMutual}, []int{})
```

Comment:
I needed to include flags.APOptionMutualRequired (in the empty int list) for AD support